fix(wallet)_: temporary improvements on token collisions problem

saledjenic opened the pull request:
What's changed:
- analyzer updated to identify symbol and decimal collisons
- before resolving the collision, the duplicate symbols are resolved within each list by skipping other entries
- if there is a symbol collision the uniswap tokens list is the source of truth
- if there is a decimal collision (the same symbol, but different decimals used) the symbol is updated (e.g. USDC(6), USDC(18))

The following screenshot describes how many collisions we have across token lists (Status, Aave, Uniswap):
![image](https://github.com/user-attachments/assets/e5540d79-f1ae-41ab-ab74-022948f20564)

saledjenic committed:
fix(wallet)_: temporary improvements on token collisions problem

What's changed:
- analyzer updated to identify symbol and decimals collisons
- before resolving the collision, the duplicate symbols are resolved within each list by skipping other entries
- if there is a symbol collision the uniswap tokens list is the source of truth
- if there is a decimal collision (the same symbol, but different decimals used) the symbol is updated (e.g. USDC(6), USDC(18))

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (55)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 91497a6f | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/1/) | 2025-04-16 18:18:23 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250416-181526-91497a-pr6538.aar) |
| :heavy_check_mark: | 91497a6f | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/1/) | 2025-04-16 18:18:59 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250416-181527-91497a-pr6538.zip) |
| :heavy_check_mark: | 91497a6f | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/1/) | 2025-04-16 18:20:43 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250416-181532-91497a-pr6538.zip) |
| :heavy_check_mark: | 91497a6f | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/1/) | 2025-04-16 18:21:25 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250416-181537-91497a-pr6538.zip) |
| :heavy_check_mark: | 91497a6f | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/1/) | 2025-04-16 18:21:39 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250416-181528-91497a-pr6538.zip) |
| :heavy_check_mark: | 91497a6f | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/1/) | 2025-04-16 18:22:23 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250416-181532-91497a-pr6538.zip) |
| :heavy_check_mark: | 91497a6f | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/1/) | 2025-04-16 18:23:55 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/1//consoleText) |
| :heavy_multiplication_x: | 91497a6f | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/1/) | 2025-04-16 18:50:56 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c101ad9f | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/2/) | 2025-04-17 08:39:12 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250417-083610-c101ad-pr6538.aar) |
| :heavy_check_mark: | c101ad9f | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/2/) | 2025-04-17 08:39:30 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250417-083611-c101ad-pr6538.zip) |
| :heavy_check_mark: | c101ad9f | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/2/) | 2025-04-17 08:41:21 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250417-083614-c101ad-pr6538.zip) |
| :heavy_check_mark: | c101ad9f | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/2/) | 2025-04-17 08:41:23 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250417-083615-c101ad-pr6538.zip) |
| :heavy_check_mark: | c101ad9f | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/2/) | 2025-04-17 08:41:57 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250417-083619-c101ad-pr6538.zip) |
| :heavy_check_mark: | c101ad9f | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/2/) | 2025-04-17 08:42:09 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250417-083612-c101ad-pr6538.zip) |
| :heavy_check_mark: | c101ad9f | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/2/) | 2025-04-17 08:45:47 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/2//consoleText) |
| :heavy_multiplication_x: | c101ad9f | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/2/) | 2025-04-17 09:11:47 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ec89daf4 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/3/) | 2025-04-17 11:08:36 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250417-110526-ec89da-pr6538.aar) |
| :heavy_check_mark: | ec89daf4 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/3/) | 2025-04-17 11:08:45 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250417-110527-ec89da-pr6538.zip) |
| :heavy_check_mark: | ec89daf4 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/3/) | 2025-04-17 11:10:01 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250417-110535-ec89da-pr6538.zip) |
| :heavy_check_mark: | ec89daf4 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/3/) | 2025-04-17 11:10:34 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250417-110530-ec89da-pr6538.zip) |
| :heavy_check_mark: | ec89daf4 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/3/) | 2025-04-17 11:10:44 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250417-110531-ec89da-pr6538.zip) |
| :heavy_check_mark: | ec89daf4 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/3/) | 2025-04-17 11:11:25 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250417-110528-ec89da-pr6538.zip) |
| :heavy_check_mark: | ec89daf4 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/3/) | 2025-04-17 11:14:59 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/3//consoleText) |
| :heavy_multiplication_x: | ec89daf4 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/3/) | 2025-04-17 11:40:45 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 361bc6fd | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/4/) | 2025-04-22 09:53:03 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250422-095006-361bc6-pr6538.aar) |
| :heavy_check_mark: | 361bc6fd | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/4/) | 2025-04-22 09:53:31 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250422-095007-361bc6-pr6538.zip) |
| :heavy_check_mark: | 361bc6fd | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/4/) | 2025-04-22 09:55:00 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250422-095015-361bc6-pr6538.zip) |
| :heavy_check_mark: | 361bc6fd | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/4/) | 2025-04-22 09:55:20 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250422-095012-361bc6-pr6538.zip) |
| :heavy_check_mark: | 361bc6fd | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/4/) | 2025-04-22 09:55:41 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250422-095010-361bc6-pr6538.zip) |
| :heavy_check_mark: | 361bc6fd | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/4/) | 2025-04-22 09:58:03 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/4//consoleText) |
| :heavy_check_mark: | 361bc6fd | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/4/) | 2025-04-22 09:58:32 | ~8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250422-095009-361bc6-pr6538.zip) |
| :heavy_multiplication_x: | 361bc6fd | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/4/) | 2025-04-22 10:25:06 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2bdf217e | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/5/) | 2025-04-23 07:15:22 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250423-071212-2bdf21-pr6538.aar) |
| :heavy_check_mark: | 2bdf217e | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/5/) | 2025-04-23 07:15:31 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250423-071214-2bdf21-pr6538.zip) |
| :heavy_check_mark: | 2bdf217e | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/5/) | 2025-04-23 07:17:08 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250423-071224-2bdf21-pr6538.zip) |
| :heavy_check_mark: | 2bdf217e | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/5/) | 2025-04-23 07:17:24 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250423-071218-2bdf21-pr6538.zip) |
| :heavy_check_mark: | 2bdf217e | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/5/) | 2025-04-23 07:17:29 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250423-071220-2bdf21-pr6538.zip) |
| :heavy_check_mark: | 2bdf217e | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/5/) | 2025-04-23 07:18:18 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250423-071215-2bdf21-pr6538.zip) |
| :heavy_multiplication_x: | 2bdf217e | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/5/) | 2025-04-23 07:47:16 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b1c99de3 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/6/) | 2025-04-23 13:44:26 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250423-134120-b1c99d-pr6538.aar) |
| :heavy_check_mark: | b1c99de3 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/6/) | 2025-04-23 13:44:49 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250423-134122-b1c99d-pr6538.zip) |
| :heavy_check_mark: | b1c99de3 | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/6/) | 2025-04-23 13:45:49 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250423-134130-b1c99d-pr6538.zip) |
| :heavy_check_mark: | b1c99de3 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/6/) | 2025-04-23 13:47:12 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250423-134125-b1c99d-pr6538.zip) |
| :heavy_check_mark: | b1c99de3 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/6/) | 2025-04-23 13:47:35 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250423-134127-b1c99d-pr6538.zip) |
| :heavy_check_mark: | b1c99de3 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/6/) | 2025-04-23 13:48:37 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250423-134123-b1c99d-pr6538.zip) |
| :heavy_check_mark: | b1c99de3 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/6/) | 2025-04-23 13:50:59 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/6//consoleText) |
| :heavy_multiplication_x: | b1c99de3 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/6/) | 2025-04-23 14:16:58 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | afda18fb | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/7/) | 2025-04-24 14:23:10 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250424-141951-afda18-pr6538.aar) |
| :heavy_check_mark: | afda18fb | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/7/) | 2025-04-24 14:23:54 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250424-141952-afda18-pr6538.zip) |
| :heavy_check_mark: | afda18fb | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/7/) | 2025-04-24 14:25:10 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250424-142001-afda18-pr6538.zip) |
| :heavy_check_mark: | afda18fb | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/7/) | 2025-04-24 14:25:46 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250424-141955-afda18-pr6538.zip) |
| :heavy_check_mark: | afda18fb | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/7/) | 2025-04-24 14:26:02 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250424-141954-afda18-pr6538.zip) |
| :heavy_check_mark: | afda18fb | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/7/) | 2025-04-24 14:27:11 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250424-141957-afda18-pr6538.zip) |
| :heavy_check_mark: | afda18fb | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/7/) | 2025-04-24 14:30:27 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/7//consoleText) |
| :heavy_multiplication_x: | afda18fb | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/7/) | 2025-04-24 14:54:09 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/7//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 5362adc2 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/8/) | 2025-04-25 06:59:39 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250425-065640-5362ad-pr6538.zip) |
| :heavy_check_mark: | 5362adc2 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/8/) | 2025-04-25 06:59:54 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250425-065639-5362ad-pr6538.aar) |
| :heavy_check_mark: | 5362adc2 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/8/) | 2025-04-25 07:00:11 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250425-065643-5362ad-pr6538.zip) |
| :heavy_check_mark: | 5362adc2 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/8/) | 2025-04-25 07:01:05 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250425-065649-5362ad-pr6538.zip) |
| :heavy_check_mark: | 5362adc2 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/8/) | 2025-04-25 07:01:54 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250425-065645-5362ad-pr6538.zip) |
| :heavy_check_mark: | 5362adc2 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/8/) | 2025-04-25 07:04:15 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250425-065641-5362ad-pr6538.zip) |
| :heavy_multiplication_x: | 5362adc2 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/8/) | 2025-04-25 07:08:51 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/8//consoleText) |
| :heavy_multiplication_x: | 5362adc2 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/8/) | 2025-04-25 07:31:03 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6538/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 773c2763 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6538/9/) | 2025-04-25 07:02:49 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250425-065944-773c27-pr6538.zip) |
| :heavy_check_mark: | 773c2763 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6538/9/) | 2025-04-25 07:03:46 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250425-065958-773c27-pr6538.aar) |
| :heavy_check_mark: | 773c2763 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6538/9/) | 2025-04-25 07:05:17 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250425-070016-773c27-pr6538.zip) |
| :heavy_check_mark: | 773c2763 | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6538/9/) | 2025-04-25 07:05:56 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250425-070115-773c27-pr6538.zip) |
| :heavy_check_mark: | 773c2763 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6538/9/) | 2025-04-25 07:07:01 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250425-070159-773c27-pr6538.zip) |
| :heavy_check_mark: | 773c2763 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6538/9/) | 2025-04-25 07:09:46 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250425-070420-773c27-pr6538.zip) |
| :heavy_multiplication_x: | 773c2763 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/9/) | 2025-04-25 07:19:16 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6538/9//consoleText) |

smohamedjavid commented:
@saledjenic Awesome work 🙌 

I did local testing, and the tokens with different decimals on different chains, such as USDC and USDT, are prevented from colliding with other chains with the help of a token symbol update with decimals. 👍 

I discovered a different issue on mobile. The prices for tokens are fetched through the `wallet_fetchPrices` RPC, which is called with token symbols, in our case, `USDC(6) USDC(18) USDT(6) USDT(18)` and there are no tokens by that symbol in our market providers (fails to 0). Is it possible to handle these tokens (sending the actual token symbol) before requesting the market provider for prices?

saledjenic commented:
@smohamedjavid yes, that's a good catch. I was initially thinking of handling that on the client, somehow, but let me think again and see what can be done.

saledjenic committed:
fix(wallet)_: low, medium, high estimated times added to the copy function for the path type

saledjenic committed:
fix(wallet)_: an extra mapping for fetching prices

Since the decimal collision is solved by adding the decimals as postfix to the symbol name
an extra mapping is necessary to map those symbols to/from symbols recognized by the price providers.

saledjenic commented:
@smohamedjavid could you try it now? I've added mapping for fetching prices. No changes needed on the client side, just point to the latest changes and test.

smohamedjavid commented:
@saledjenic - I checked and all good 🙌

dlipicar committed:
fix(wallet)_: fixed pegged token currency format

saledjenic committed:
fix(wallet)_: update tokens' postfix from decimals to l1 chain based mark

Changes include:
- Replaced postfix for decimals collision, symbols include l1 chain based mark instead of decimals.
- Token filtering logic added to ensure that only chains the app knows about are present (otherwise, it cannot guarantee uniqueness without decimals included).
- Replaced hardcoded chain ID with a constant for `TestnetChainID` and added to `AllChainIDs`, which fixed tests balancechecker.
- Updated no-balance tests for the router that were failing due to the new BSC chain added.

saledjenic commented:
@alaibe @shivekkhurana a new commit has been added to this PR, it's about updating tokens' postfix from decimals to L1 chain-based mark.

Based on the discussion, we can merge this to https://github.com/status-im/status-go/pull/6466 and continue the work from a single place.

One unresolved issue is BNB native vs BNB in ethereum chain token. Need to discuss how to fix it.
```
            {
              "address": "0x0000000000000000000000000000000000000000",
              "name": "BNB",
              "symbol": "BNB",
              "decimals": 18,
              "chainId": 56,
              "pegSymbol": "",
              "verified": true
            },
            {
              "address": "0x0000000000000000000000000000000000000000",
              "name": "BNB",
              "symbol": "BNB",
              "decimals": 18,
              "chainId": 97,
              "pegSymbol": "",
              "verified": true
            },
```

```
            {
              "address": "0xb8c77482e45f1f44de1745f52c74426c631bdd52",
              "name": "BNB",
              "symbol": "BNB",
              "decimals": 18,
              "chainId": 1,
              "pegSymbol": "",
              "verified": false
            },
```
